index/src/lib.rs

```diff
@@ -330,15 +330,19 @@ impl IndexWriter {
         match index.parse_doc(data) {
             Ok(doc) => {
                 let id = &id(&doc);
-                let doc = index.index_doc(id, &doc).map_err(|e| {
-                    self.metrics.failed_total.inc();
-                    e
-                })?;
-                self.delete_document(index, id);
-                self.writer.add_document(doc).map_err(|e| {
+                let docs = index.index_doc(id, &doc).map_err(|e| {
                     self.metrics.failed_total.inc();
                     e
                 })?;
+                for (i, doc) in docs {
+                    self.delete_document(index, &i);
+                    self.writer.add_document(doc).map_err(|e| {
+                        let e = e.clone();
```

Comment:
Is this clone required? or could the function simply return `e`?

Reply:
if do not use e.clone(), the code will have the error "Use of moved value ".

Reply:
Hm, I don't see where it could be moved. Maybe some old code that was there before?

Reply:
it's intellij's mentioned error, after remove it, it's work. I will remove it.